### DISCUSSION TITLE
Documentation for `Stream`

### DISCUSSION THREAD

**Discussion by Z-snails** [zesterer/chumsky#83] [General]
What should `eoi` be in `Stream::from_iter`?

**Comment by zesterer** [comment]
`eoi` should be an 'End Of Input' span, i.e: a span that might be used when there are no items left in the stream. It's required for things like `map_with_span` to work properly for parsers that consume no input, among other things.

As an example, the default implementation for `&str` just uses `str.len()..str.len()` (i.e: a zero-width span starting and ending at the end of the string).
